src/v/transform/transform_processor.cc

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "transform/transform_processor.h"
+
+#include "model/record_batch_reader.h"
+#include "model/timeout_clock.h"
+#include "random/simple_time_jitter.h"
+#include "transform/logger.h"
+#include "wasm/api.h"
+
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/queue.hh>
+#include <seastar/core/sleep.hh>
+
+namespace transform {
+
+namespace {
+model::record_batch_reader::data_t
+extract_batches(model::record_batch_reader::storage_t s) {
```

Comment:
is this the same as calling `model::consume_reader_to_memory` on the reader?

Reply:
Yes, this looks funny because it was adapted from a more complex thing that tried to minimize latency by:

1) never waiting to read all batch of batches before starting transforms
2) pipelining so that we were concurrently fetching while doing transforms.